### PR TITLE
Collect ODF Provider and Consumer cluster resources

### PR DESCRIPTION
**Pull request by Mrudraia**
This PR will add additional  missing resources in ODF provider cluster and consumer cluster. In provider  cluster  the following resources YAMLs, OC_OUPUTs, OC_DESC are collected.

- storage consumers
- storage class requests

In Consumer cluster

- storageclients 
- storageclassclaims 
- ocs-client-operator logs, pods, api-server logs etc
- ceph command outputs

### CODE REVIEW

File: collection-scripts/gather [hunk 1]
```diff
@@ -38,6 +40,7 @@ elif [ "${reconcileStrategy}" = "standalone" ]; then
 else
     dbglog "Collecting entire ceph logs"
     gather_ceph_resources ${BASE_COLLECTION_PATH}
+    gather_odf_consumer ${BASE_COLLECTION_PATH}
```
Comment (leelavg) [inline collection-scripts/gather:43]: - does above command `ceph_resources` fail silently
- the log will say `ceph logs` but you'll not access ceph via cli at consumer end
- btw, let's use the naming `client` inplace of `consumer`

Comment (leelavg) [inline collection-scripts/gather:43]: - there are scenarios where a single cluster will run in both provider & client modes and you'll be gathering only provider MG in that scenario.

File: collection-scripts/gather-odf-consumer [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+# Expect base collection path as an exported variable
+# If it is not defined, use PWD instead
+BASE_COLLECTION_PATH=${BASE_COLLECTION_PATH:-"$(pwd)"}
+
+
+STORAGE_CLIENT_NAMESPACE=$(oc get storageclient -A --no-headers | awk '{print $1}')
+INSTALL_NAMESPACE="openshift-storage"
```
Comment (leelavg) [inline collection-scripts/gather-odf-consumer:9]: - what's the diff b/n `storage_client_ns` and `install_ns` here?

File: collection-scripts/gather-odf-consumer [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+commands_desc+=("pods")
+commands_get+=("subscription")
+commands_desc+=("storageclients")
+commands_desc+=("storageclassclaims")
+
+
+for INSTALL_NAMESPACE in  $STORAGE_CLIENT_NAMESPACE $INSTALL_NAMESPACE ; do
+     dbglog "collecting dump of namespace ${INSTALL_NAMESPACE}"
+     { oc adm --dest-dir="${BASE_COLLECTION_PATH}" inspect ns/"${INSTALL_NAMESPACE}" --"${SINCE_TIME}" 2>&1; } | dbglog
+     dbglog "collecting dump of clusterresourceversion"
```
Comment (leelavg) [inline collection-scripts/gather-odf-consumer:40]: what is clusterresourceversion?

File: collection-scripts/gather-odf-consumer [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+     for oc_yaml in "${oc_yamls[@]}"; do
+          # shellcheck disable=SC2129
+          { oc adm --dest-dir="${BASE_COLLECTION_PATH}" inspect "${oc_yaml}" -n "${INSTALL_NAMESPACE}" --"${SINCE_TIME}" 2>&1; } | dbglog
```
Comment (leelavg) [inline collection-scripts/gather-odf-consumer:43]: - iirc, client-operator is cluster scoped operator with storageclient crd at namespace scope and storageclassclaims crd at cluster scope
- as such, you should collect claims and client CRs from where ever they present however, should only collect operator related info (pods, logs, subs etc) from wherever client-operator is running only

File: collection-scripts/gather-odf-consumer [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+     done
+
+     # Run the Collection of OC desc commands
+     for command_desc in "${commands_desc[@]}"; do
+          dbglog "collecting oc describe command ${command_desc}"
+          COMMAND_OUTPUT_FILE=${BASE_COLLECTION_PATH}/namespaces/${INSTALL_NAMESPACE}/oc_output/${command_desc// /_}
+          # shellcheck disable=SC2086
+          { oc describe ${command_desc} -n ${INSTALL_NAMESPACE}; } >>"${COMMAND_OUTPUT_FILE}"
+     done
+done
```
Comment (leelavg) [inline collection-scripts/gather-odf-consumer:64]: add newline to the end (same for other files)

File: collection-scripts/gather-odf-provider [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+# Expect base collection path as an exported variable
+# If it is not defined, use PWD instead
+BASE_COLLECTION_PATH=${BASE_COLLECTION_PATH:-"$(pwd)"}
+
+INSTALL_NAMESPACES=$(oc get storagecluster -A --no-headers | awk '{print $1}')
+
+# collection path for OC commands
+mkdir -p "${BASE_COLLECTION_PATH}/odf-provider-cluster/oc_output/"
+
+# Command List
```
Comment (leelavg) [inline collection-scripts/gather-odf-provider:12]: you should also be adding `storageprofile` to the lists and at the same time if they don't exist the commands should fail safely (if not the same already)

File: collection-scripts/gather-odf-consumer (hunk 4 above)
Comment (Madhu-1) [reply collection-scripts/gather-odf-consumer:43]: Yes that's correct.

Comment (leelavg) [reply collection-scripts/gather-odf-consumer:43]: ack, thanks for the confirmation.

File: collection-scripts/gather (hunk 1 above)
Comment (Mrudraia) [reply collection-scripts/gather:43]: Yes ceph_resources will fail silently.
renamed client with consumer

Comment (Mrudraia) [reply collection-scripts/gather:43]: For such scenarios, the check for storagecluster does not work. We might need to change the if else logic conditions

**Comment by Mrudraia** [comment]
this PR is closed due to changes in work flow

**Pull request closed without merging**